tools/firtool/firtool.cpp

```diff
@@ -658,6 +658,8 @@ processBuffer(MLIRContext &context, TimingScope &ts, llvm::SourceMgr &sourceMgr,
     pm.nest<firrtl::CircuitOp>().nest<firrtl::FModuleOp>().addPass(
         firrtl::createFlattenMemoryPass());
 
+  if (!disableCheckCombCycles)
+    pm.nest<firrtl::CircuitOp>().addPass(firrtl::createCheckCombLoopsPass());
   // The input mlir file could be firrtl dialect so we might need to clean
   // things up.
```

Comment:
If you want to run the pass before LowerType, it's necessary to handle subaccess used as sink.